actionview/lib/action_view/helpers/tag_helper.rb

```diff
@@ -68,12 +83,12 @@ def p(*arguments, **options, &block)
 
         def tag_string(name, content = nil, escape: true, **options, &block)
           content = @view_context.capture(self, &block) if block_given?
-          self_closing = SVG_SELF_CLOSING_ELEMENTS.include?(name)
-          if (HTML_VOID_ELEMENTS.include?(name) || self_closing) && content.nil?
-            "<#{name.to_s.dasherize}#{tag_options(options, escape)}#{self_closing ? " />" : ">"}".html_safe
-          else
-            content_tag_string(name.to_s.dasherize, content || "", options, escape)
-          end
+
+          content_tag_string(name.to_s.dasherize, content, options, escape)
+        end
+
+        def void_tag_string(name, options, escape = true, self_closing = false)
```

Comment:
Do we need this one? I think if we inline it will be faster

Reply:
I've benchmarked with the string building inline:

```
❯ ruby bench.rb
Warming up --------------------------------------                               
                 tag    73.192k i/100ms                                                                                                                                               
         tag_builder    74.687k i/100ms
Calculating -------------------------------------
                 tag    739.532k (± 1.1%) i/s -      3.733M in   5.048096s
         tag_builder    753.460k (± 1.2%) i/s -      3.809M in   5.056087s
                                             
Comparison:
         tag_builder:   753460.1 i/s                                                       
                 tag:   739532.5 i/s - same-ish: difference falls within error
```

and with the method definition:

```
❯ ruby bench.rb
Warming up --------------------------------------
                 tag    74.378k i/100ms
         tag_builder    81.684k i/100ms
Calculating -------------------------------------
                 tag    744.862k (± 0.3%) i/s -      3.793M in   5.092651s
         tag_builder    817.567k (± 0.9%) i/s -      4.166M in   5.095849s

Comparison:
         tag_builder:   817566.9 i/s
                 tag:   744862.0 i/s - 1.10x  (± 0.00) slower
```

For whatever reason, the method invocation is slightly more performant.

Reply:
Ah, probably because of the method case. It doesn't matter with YJIT either, so we can continue using the private method.